svm/src/transaction_processing_callback.rs

```diff
@@ -7,4 +7,16 @@ pub trait TransactionProcessingCallback {
     fn get_account_shared_data(&self, pubkey: &Pubkey) -> Option<AccountSharedData>;
 
     fn add_builtin_account(&self, _name: &str, _program_id: &Pubkey) {}
+
+    fn inspect_account(&self, _address: &Pubkey, _account_state: AccountState, _is_writable: bool) {
+    }
+}
+
+/// The state the account is in initially, before transaction processing
+#[derive(Debug)]
+pub enum AccountState<'a> {
```

Comment:
I think it is fine to add the Dead variant for "new account" to support a more general inspection function. 
But for LTHash, we probably only need to care about Alive accounts. "new" account will be treated as 0 for LTHash. We don't need to put them in the cache. 
https://github.com/brooksprumo/solana/blob/755ccd379e5ad3a2ee241406feae673963277e18/runtime/src/bank.rs#L5220-L5253

Reply:
In the inspect callback for lthash, we cache the original version of the account in a map, 

In the block, when an account is stored, we add the new accounts hash and subtract the old account hash if it is found in the cache. We don't load the previous version of the account before store. 

I think that's why we introduce the cache for lthash when we do transaction loading. This avoids to load the accounts twice in the block.

Reply:
but i think tracking dead accounts is a good idea, which may be useful for other inspection cases.

Reply:
I think the intention makes sense, but I'm wondering if these variant names are the best. You guys know accounts better than I do, but my mind goes toward something like `DoesNotExist` and `Exists`.

Is there somewhere we use `Dead` and `Alive` specifically for accounts already? I didn't find anything in AccountsDB from a quick search.

Reply:
Sounds good to me, I think it's fine as-is if it's a naming convention you guys use in AccountsDB.